sa/type-converter.go

```diff
@@ -38,6 +38,16 @@ func (tc BoulderTypeConverter) ToDb(val interface{}) (interface{}, error) {
 		return string(t), nil
 	case core.OCSPStatus:
 		return string(t), nil
+	case *core.AcmeURL:
```

Comment:
So I'm still of the opinion these need type converters. I'm not sure what you mean by `core.AcmeURL + jose.JsonWebSignature` shouldn't be in a struct passed to gorp. If you mean that the `Challenge` struct fields should be changed to `string` and `[]byte` respectively and that the callers should deal with un/marshalling these fields to the correct types then I'm not sure I'm convinced.

`jose.JsonWebSignature` in particular doesn't have `Un/MarshalJSON` methods so gorp won't properly move it in/out automatically (hence it throwing errors without a type converter), in the case where the caller converts them then that is essentially what I am already doing under the hood with the type converter, moving repetitive work away from the caller.

Also `Validation` should now properly be a blob, I think the rest of the fields should be correct as well.